subsys/bluetooth/mesh/gen_lvl_srv.c

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <string.h>
+#include <bluetooth/mesh/gen_lvl_srv.h>
+#include <bluetooth/mesh/gen_dtt_srv.h>
+#include "model_utils.h"
+
+static void encode_status(const struct gen_lvl_status *status,
+			  struct net_buf_simple *buf)
+{
+	net_buf_simple_add_le16(buf, status->present);
+	if (status->present != status->target) {
```

Comment:
I think it is better to have separate status encode function for the `move_set` so as to report the correct status for this message.

Reply:
Hmm, in the section about move_set the spec says this:

> The target Generic Level state is the upper limit of the Generic Level state when the transition speed is positive, or the lower limit of the Generic Level state when the transition speed is negative.

Currently, it's up to the application to follow this. To change that, we'll have to introduce an internal state for the type of transition we're doing, and keep that in sync with the timing. IMO it's better to just document this requirement in the API, as this potentially adds quite a bit of complexity, and breaks the consistency of the API.

Reply:
Ok, I think documenting this should be fine. However, then, as a user of this API, gen_plvl_srv will have to compose a correct status as mentioned here: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1238#discussion_r333978484

Reply:
resolved.